contracts/AssetPool.sol

```diff
@@ -64,11 +96,122 @@ contract AssetPool is Ownable {
         _deposit(msg.sender, amount);
     }
 
-    /// @notice Withdraws collateral from the asset pool. Accepts the amount of
-    ///         underwriter tokens representing the share of the pool that
-    ///         should be withdrawn.
+    /// @notice Initiates the withdrawal of collateral and rewards from the pool.
+    ///         Accepts the amount of underwriter tokens representing the share
+    ///         of the pool that should be withdrawn.
+    ///         Underwriter needs to complete the withdrawal by calling the
+    ///         `completeWithdrawal` function after the withdrawal delay passes
+    ///         but before the graceful withdrawal timeout ends to avoid part of
+    ///         their share being seized by the pool.
     /// @dev Before calling this function, underwriter token needs to have the
     ///      required amount accepted to transfer to the asset pool.
+    function initiateWithdrawal(uint256 covAmount) external {
+        uint256 covBalance = underwriterToken.balanceOf(msg.sender);
+        require(
+            covAmount <= covBalance,
+            "Underwriter token amount exceeds balance"
+        );
+        require(
+            covAmount > 0,
+            "Underwriter token amount must be greater than 0"
+        );
+
+        underwriterToken.safeTransferFrom(msg.sender, address(this), covAmount);
+
+        pendingWithdrawal[msg.sender] = covAmount.add(
+            pendingWithdrawal[msg.sender]
+        );
+        /* solhint-disable not-rely-on-time */
+        withdrawalInitiatedTimestamp[msg.sender] = block.timestamp;
```

Comment:
just to make sure... If I'm an underwriter and I initiate my first withdrawal today for amount of 1000 and a week after I initiate a second withdrawal for amount of 1, then technically I will be able to withdraw my tokens 2 weeks after the second initiation (3 weeks after the first initiation), right? Basically, the latest initiation overrides whatever was initiated earlier?

Reply:
Correct.